### PR TITLE
MINOR:  kafka-run-class.sh now runs under Cygwin.

### PR DESCRIPTION
**Pull request by deragon**
kafka-run-class.sh now runs under Cygwin; paths and classpath are set up properly.

**WARNING:**  The script was not tested on a Linux machine, only under Cygwin.  Prior to merge it into trunk, if accepted, please run a quick test to ensure nothing broke.  From my own code review, there should not be any problem, but we can never be too sure.

I do not have the environment to test it under Linux at this moment.

### CODE REVIEW

File: bin/kafka-run-class.sh [hunk 1]
```diff
@@ -20,6 +20,10 @@ then
   exit 1
 fi
 
+# CYGINW == 1 if Cygwin is detected, else 0.
+[[ $(uname -a) =~ "CYGWIN" ]]
+CYGWIN=$((! $?))
```
Comment (ewencp) [inline bin/kafka-run-class.sh]: This is just an aspiration, but given that we've had issues with scripts in the past it'd be nice to structure them such that we could turn on `set -e` to more aggressively fail if we encounter issues. I think this test is not safe in that context since it runs the command and then uses `$?` instead of using the command in an if. Could we just turn this into an if or a one-liner to make it safe if we decided to enable `set -e`?

File: bin/kafka-run-class.sh [hunk 2]
```diff
@@ -145,20 +149,25 @@ fi
...
+  LOG_DIR="$base_dir/logs"
 fi
 
 # Log4j settings
 if [ -z "$KAFKA_LOG4J_OPTS" ]; then
   # Log to console. This is a tool.
-  KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/config/tools-log4j.properties"
+  LOG4J_DIR="$base_dir/config/tools-log4j.properties"
+  # If Cygwin is detected, LOG4J_DIR is converted to Windows format.
+  (( CYGWIN )) && LOG4J_DIR=`cygpath --path --mixed "${LOG4J_DIR}"`
```
Comment (ewencp) [inline bin/kafka-run-class.sh]: Can we use `$(...)` instead of backticks throughout this script? `shellcheck` complains about legacy backticks

**Comment by ewencp** [comment]
@deragon Left a few minor comments that should be easy to clean up. I'm open to merging this, but I'll want to also trigger a full system test run before doing so. We've recently run into quite a few compatibility issues with changes to the shell scripts and they don't have any testing besides the system tests, so we're treading carefully with any new changes. Once we've got a completed patch in place I'd like to validate on my own Windows VM, and the system tests should hopefully give good coverage on Linux.

Also, can you fix up the PR title to match the naming suggested in the [contribution guidelines](https://cwiki.apache.org/confluence/display/KAFKA/Contributing+Code+Changes#ContributingCodeChanges-PullRequest).

**Comment by ewencp** [comment]
I started a system test build against the current patch [here](https://jenkins.confluent.io/job/system-test-kafka-branch-builder/467/). We should follow up on the final patch, but since the comments I had were minor, I figure it's worth testing against the current version to catch any issues asap.

**Comment by deragon** [comment]
Ok, I brought the changes requested.  Let me know if I can do anything more.

**Comment by deragon** [comment]
Any chances to have my pull requested accepted?

File: bin/kafka-run-class.sh [hunk 3]
```diff
@@ -41,11 +48,11 @@ should_include_file() {
 base_dir=$(dirname $0)/..
 
 if [ -z "$SCALA_VERSION" ]; then
-	SCALA_VERSION=2.10.6
+  SCALA_VERSION=2.10.6
```
Comment (ewencp) [inline bin/kafka-run-class.sh:51]: seems like we have a bunch of unrelated whitespace changes in this patch

**Comment by ewencp** [comment]
@deragon Have kicked off https://jenkins.confluent.io/job/system-test-kafka-branch-builder/492 just to check that we don't break anything non-Cygwin. If we can clean up unrelated changes and the system tests look good, I think we can merge this.

**Comment by deragon** [comment]
The unrelated change is simply a lonely tab that was replaced for spaces, to follow the rest of the structure of the files.  Its a slight improvement.  Please leave it there; everybody hates inconsistent structure.

**Comment by ewencp** [comment]
@granders @ijuma Someone else want to give this a quick once over? Things look ok, but given our previous luck with the shell scripts...

**Comment by ijuma** [comment]
My knowledge of cygwin is close to 0, so hopefully @granders can help. :)

**Comment by deragon** [comment]
At least, ensure that it works under Linux, i.e. I broke nothing. If for some odd reason, my solution on Cygwin would have only worked on my Workstation, well the script would not be worse of than before since it never worked in Cygwin anyhow. 

That said, I am confident of my patch, but I understand totally your caution and applaud your process.

**Comment by ewencp** [comment]
@ijuma I even just meant careful review of the bash in the context of not breaking anything on other platforms :) LGTM
